Fixes part of #10700: Refactor AnswerStatsObjectFactory to model.ts

parulpriyedarshani opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #10700.
2. This PR does the following: Refactoring the AnswerStatsObjectFactory to model.ts file in Angular.

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The linter/Karma presubmit checks have passed locally on your machine.
- [ ] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [ ] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes made in this PR work correctly.
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Oppiabot can assign anyone for review/help if you leave a comment like the following: "{{Question/comment}} @{{reviewer_username}} PTAL"
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

Commit:
Refactoring AnswerStatsObjectFactory

Commit:
Refactoring AnswerStatsObjectFactory

parulpriyedarshani commented:
@vojtechjelinek @srijanreddy98 PTAL!

parulpriyedarshani commented:
Could anyone please take a look at the PR. 
Thanks

nishantwrp reviewed (approved):
codeowner files, lgtm! Thanks.

vojtechjelinek reviewed (approved):
LGTM!

brianrodri reviewed (changes requested):
Please make the filename `answer-stats.model.ts`

Commit:
Merge branch 'refactor' of https://github.com/parulpriyedarshani/oppia into refactor

Commit:
Merge branch 'refactor' of https://github.com/parulpriyedarshani/oppia into refactor

Commit:
Merge branch 'refactor' of https://github.com/parulpriyedarshani/oppia into refactor

Commit:
Merge branch 'refactor' of https://github.com/parulpriyedarshani/oppia into refactor

parulpriyedarshani commented:
The lint checks are failing to throw the error: Files in tsconfig-strict.json are not alphabetically sorted. But I think that the order is fine. Can anyone please take a look into this?

nishantwrp commented:
@parulpriyedarshani, I think it's because 'a' comes after 'E'. (Notice that 'E' is in upper case). Eg. When you do, `['a', 'E'].sort()` 'a' gets placed after 'E'.

brianrodri commented on core/templates/domain/exploration/answerstats-object.model.spec.ts:
```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
```
Looks like some cruft made it into the PR, please delete this file.

brianrodri commented on core/templates/domain/exploration/answerstats-object.model.ts:
```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
```
Ditto, please delete this file.

kevintab95 reviewed (changes requested):
Hi @parulpriyedarshani, I have the same comments as Brian for the codeowner files. Please reassign me once you have addressed those comments.

Also, I'm a bit surprised that these changes were not caught locally. Did the lint checks not fail when you tried to push changes? If it didn't, it is worth investigating that first.

srijanreddy98 commented on core/templates/App.ts:
```diff
@@ -98,7 +98,7 @@ angular.module('oppia').config([
       'AlertsService', 'AngularNameService',
       'AnswerClassificationResultObjectFactory', 'AnswerClassificationService',
-      'AnswerGroupObjectFactory', 'AnswerStatsObjectFactory',
+      'AnswerGroupObjectFactory', 'AnswerStats',
```
You will have to remove it from here.

parulpriyedarshani replied:
It's done! 👍

srijanreddy98 commented on core/templates/domain/exploration/answer-stats.model.spec.ts:
```diff
@@ -13,27 +13,21 @@
 
-import { AnswerStatsObjectFactory } from
-  'domain/exploration/AnswerStatsObjectFactory';
+import { AnswerStats } from
+  'domain/exploration/answer-stats.model';
```
```suggestion
import { AnswerStats } from 'domain/exploration/answer-stats.model';
```

parulpriyedarshani replied:
Done!

srijanreddy98 reviewed (changes requested):
Took a pass @parulpriyedarshani. A couple of comments otherwise LGTM!

parulpriyedarshani commented:
> Hi @parulpriyedarshani, I have the same comments as Brian for the codeowner files. Please reassign me once you have addressed those comments.
> 
> Also, I'm a bit surprised that these changes were not caught locally. Did the lint checks not fail when you tried to push changes? If it didn't, it is worth investigating that first.

I too am wondering because when I pushed the code, the lint checks failed only for the tsconfig-strict.json file for not being alphabetically sorted. But there were no errors regarding the answerstats-object.model.ts file due to which I was unaware that it still exists. I think it should be given a look.
Thanks

parulpriyedarshani commented:
> @parulpriyedarshani, I think it's because 'a' comes after 'E'. (Notice that 'E' is in upper case). Eg. When you do, `['a', 'E'].sort()` 'a' gets placed after 'E'.

Thanks a lot! 👍

Commit:
Merge branch 'refactor' of https://github.com/parulpriyedarshani/oppia into refactor

Commit:
Merge branch 'refactor' of https://github.com/parulpriyedarshani/oppia into refactor

parulpriyedarshani commented:
@kevintab95 @srijanreddy98 The requested changes have been made. PTAL!

kevintab95 reviewed (approved):
LGTM for codeowner files. Thanks @parulpriyedarshani!

srijanreddy98 reviewed (approved):
LGTM! Thanks @parulpriyedarshani.

iamprayush commented:
Hey @parulpriyedarshani, the backend and lint tests are failing, PTAL, thanks!

Commit:
Merge branch 'refactor' of https://github.com/parulpriyedarshani/oppia into refactor

parulpriyedarshani committed:
Merge branch 'develop' into refactor

parulpriyedarshani committed:
Update tsconfig-strict.json

parulpriyedarshani commented:
Can someone please start the CircleCI test?
Thanks :)

parulpriyedarshani commented:
@srijanreddy98 All the review comments have been addressed. Can we merge the PR?

sajalasati commented:
Hi @parulpriyedarshani ! Can you follow oppiabot's [comment](https://github.com/oppia/oppia/pull/11691#issuecomment-773687893) above and try updating your branch with develop? Also since the PR is approved to be merged, if the e2e tests fail again due to flakiness, you leave a comment here asking to restart them, so that your PR is merged soon :)

parulpriyedarshani commented:
> Hi @parulpriyedarshani ! Can you follow oppiabot's [comment](https://github.com/oppia/oppia/pull/11691#issuecomment-773687893) above and try updating your branch with develop? Also since the PR is approved to be merged, if the e2e tests fail again due to flakiness, you leave a comment here asking to restart them, so that your PR is merged soon :)

Sure. Thanks :)

kevintab95 commented:
Hi @parulpriyedarshani, the e2e tests are not running because there is an error in the setup step which includes typescript checks (see [this](https://app.circleci.com/pipelines/github/oppia/oppia/18490/workflows/2139abc4-c920-4ebd-9c7d-da32601d7be1/jobs/104446)). There also seem to be quite a few frontend failures (see [this](https://github.com/oppia/oppia/pull/11691/checks?check_run_id=1860540825)). Please fix these. Let us know if you need any help with this. Thanks!

kevintab95 reviewed (changes requested):
Leaving a blocking review because test failures need to be handled before this can go in.

parulpriyedarshani commented:
> Hi @parulpriyedarshani, the e2e tests are not running because there is an error in the setup step which includes typescript checks (see [this](https://app.circleci.com/pipelines/github/oppia/oppia/18490/workflows/2139abc4-c920-4ebd-9c7d-da32601d7be1/jobs/104446)). There also seem to be quite a few frontend failures (see [this](https://github.com/oppia/oppia/pull/11691/checks?check_run_id=1860540825)). Please fix these. Let us know if you need any help with this. Thanks!

Sure, will fix them.
Thanks

parulpriyedarshani commented:
@kevintab95 This happened after I tried to resolve the merge conflict. In that branch, the code used the 'AnswerStatsObjectFactory' which in this PR will be changed to 'AnswerStats'. I am unable to understand what might be the reason behind that because I didn't change the code just changed 'AnswerStatsObjectFactory' to 'AnswerStats' and it failed and now even after changing 'AnswerStats' to 'AnswerStatsObjectFactory' (which was the original one in the other branch) the karma tests are failing. Could you please guide me on this? 
Thanks

parulpriyedarshani committed:
Merge branch 'develop' into refactor

parulpriyedarshani committed:
Merge branch 'develop' into refactor

DubeySandeep commented:
@parulpriyedarshani, is this PR ready for review?

Commit:
Merge branch 'refactor' of https://github.com/parulpriyedarshani/oppia into refactor

parulpriyedarshani commented:
> @parulpriyedarshani, is this PR ready for review?

Actually, I'm not getting why the circleci & e2e tests are failing. :( Any idea? 
Thanks.

parulpriyedarshani commented:
@kevintab95 Any idea why the e2e and circleci tests are failing?

sajalasati commented:
> @kevintab95 Any idea why the e2e and circleci tests are failing?

@parulpriyedarshani Please make sure the correct person has been assigned to the PR. Otherwise they wouldn't know they have a pass on the PR.

Tip - Leaving a `PTAL @reviewer` styled comment will automatically assign the reviewer via oppiabot.

kevintab95 commented:
Restarted the test because it looks like a flake. Thanks!

kevintab95 commented:
Hi @parulpriyedarshani can you please try updating your branch with develop? Thanks!

sajalasati commented:
Hi @parulpriyedarshani ! Please note that you have some pending action on this PR. See above comments for more details. Thanks!

Commit:
Merge branch 'develop' into refactor

parulpriyedarshani commented:
> Hi @parulpriyedarshani ! Please note that you have some pending action on this PR. See above comments for more details. Thanks!

Thanks, @sajalasati :). Done!

parulpriyedarshani commented:
> Hi @parulpriyedarshani can you please try updating your branch with develop? Thanks!

PTAL @kevintab95

darksun27 reviewed (approved):
LGTM! Thanks, @parulpriyedarshani

parulpriyedarshani commented:
@sajalasati PTAL. I tried updating my branch with develop but somehow e2e tests are failing. Could you help me out with this? Thanks.

sajalasati commented:
> @sajalasati PTAL. I tried updating my branch with develop but somehow e2e tests are failing. Could you help me out with this? Thanks.

@parulpriyedarshani I've restarted the failing tests!

parulpriyedarshani commented:
> > @sajalasati PTAL. I tried updating my branch with develop but somehow e2e tests are failing. Could you help me out with this? Thanks.
> 
> @parulpriyedarshani I've restarted the failing tests!

Thanks a lot! But still, a circleci test is failing :(. What do you think might be the reason behind this so that it can be fixed.

sajalasati commented:
> > > @sajalasati PTAL. I tried updating my branch with develop but somehow e2e tests are failing. Could you help me out with this? Thanks.
> > 
> > 
> > @parulpriyedarshani I've restarted the failing tests!
> 
> Thanks a lot! But still, a circleci test is failing :(. What do you think might be the reason behind this so that it can be fixed.

@parulpriyedarshani I am not sure, all 3 of them looked like a flake to me. I've restarted the failing one again, but if it further fails you can ask @U8NWXD to look into it. But unless these failing e2e tests are blocking your PR from getting merged, you need not worry about them.

U8NWXD commented:
@kevintab95 PTAL. Have your comments been addressed? If not, they may have been buried in the thread. Could you comment a link to them for easy access?

kevintab95 reviewed (approved):
LGTM, thanks @parulpriyedarshani!
